examples/DrawingInTheAir/DrawingInTheAir.ino

```diff
@@ -0,0 +1,237 @@
+ * it will fit into a neuron, while preserving as much of the original pattern
+ * as possible. Assuming there will typically be 1-2 seconds worth of
+ * accelerometer data at 200Hz, we will need to throw away over 90% of it to
+ * meet that goal!
+ *
+ * This is done in 2 ways:
+ *
+ * 1. Each sample consists of 3 signed 16-bit values (one each for X, Y and Z).
+ *    Map each 16 bit value to a range of 0-255 and pack it into a byte,
+ *    cutting sample size in half.
+ *
+ * 2. Undersample. If we are sampling at 200Hz and the button is held for 1.2
+ *    seconds, then we'll have ~240 samples. Since we know now that each
+ *    sample, once compressed, will occupy 3 of our neuron's 128 bytes
+ *    (see #1), then we know we can only fit 42 of those 240 samples into a
+ *    single neuron (128 / 3 = 42.666). So if we take (for example) every 5th
+ *    sample until we have 42, then we should cover most of the sample window
+ *    and have some semblance of the original pattern. */
+void undersample(uint8_t *samples, int numSamples, uint8_t *vector)
+{
+    unsigned int vi = 0;
+    unsigned int si = 0;
+    unsigned int step = numSamples / samplesPerVector;
+    unsigned int remainder = numSamples - (step * samplesPerVector);
+
+    /* Centre sample window */
+    samples += (remainder / 2) * 3;
+    for (unsigned int i = 0; i < samplesPerVector; ++i) {
+        for (unsigned int j = 0; j < 3; ++j) {
+            vector[vi + j] = movingAvg(samples, numSamples, si + j, step);
+        }
+
+        si += (step * 3);
+        vi += 3;
+    }
+}
+
+void vectorFromIMUData(uint8_t *vector)
+{
+    uint8_t accel[sensorBufSize];
+    int raw[3];
+
+    unsigned int samples = 0;
+    unsigned int i = 0;
+
+    /* Wait until button is pressed */
+    while (digitalRead(buttonPin) == HIGH);
+
+    /* While button is being held... */
+    while (digitalRead(buttonPin) == LOW) {
+        if (CurieIMU.accelDataReady()) {
+
+            CurieIMU.readAccelerometer(raw[0], raw[1], raw[2]);
+
+            /* Map raw values to 0-255 */
+            accel[i] = (uint8_t) map(raw[0], IMULow, IMUHigh, 0, 255);
+            accel[i + 1] = (uint8_t) map(raw[1], IMULow, IMUHigh, 0, 255);
+            accel[i + 2] = (uint8_t) map(raw[2], IMULow, IMUHigh, 0, 255);
+
+            i += 3;
+            ++samples;
+
+            /* If there's not enough room left in the buffers
+            * for the next read, then we're done */
+            if (i + 3 > sensorBufSize) {
+                break;
+            }
+        }
+    }
+
+    undersample(accel, samples, vector);
+}
+
+void trainLetter(char letter, unsigned int repeat)
+{
+    unsigned int i = 0;
+
+    while (i < repeat) {
+        uint8_t vector[vectorNumBytes];
+
+        if (i) Serial.println("And again...");
+
+        vectorFromIMUData(vector);
+        CuriePME.learn(vector, vectorNumBytes, letter - upperStart);
+
+        Serial.println("Got it!");
+        delay(250);
```

Comment:
I actually had it at 1000 initially, but after testing it a billion times I was getting annoyed at waiting a full second between letters. You're right though, a full second is better for first-time users.